src/fabric/interval.rs

```diff
@@ -41,8 +41,7 @@ pub struct Material {
 pub struct Interval {
     pub alpha_index: usize,
     pub omega_index: usize,
-    pub role: Role,
-    pub material: Material,
+    pub material: usize,
```

Comment:
```suggestion
    pub material_id: usize,
```
But maybe that's just me